### ISSUE TITLE
Unclear what PermissionError does

### ISSUE THREAD

**Issue by cristan** [jordond/compass#102]
It's unclear what PermissionError does. The docs say:

> Geocoding operation failed because of a permission error.
>
> Either the user denied the permission, or the permission was denied forever.

But that's already what `PermissionDenied` is for. It even has a `forever` boolean.

So let's look at the code. It's caused by either one of these exceptions:
* `PermissionMissingException`
  * This isn't used at all, thus it can simply be deleted.
* `PermissionRequestException`
  * This only happens when the WeakReference requester is null at `ActivityProvider`. Seems like very much an unexpected error.

I would delete `PermissionMissingException`, `PermissionRequestException` and the entire `PermissionError`. In order to do so, I would throw an `IllegalStateException` in `ActivityProvider` instead. This makes your library quite a bit easier: right now it's quite confusing to see the difference between `PermissionError` and `PermissionDenied`.

**Comment by jordond** [comment]
I agree, again this is a case of an earlier design of the libraries API requiring this. Then getting refactored away later on.

**Comment by jordond** [comment]
Fixed in 2.0.0
